### PR TITLE
git-utimes: Change files modification time to their last commit date

### PR DESCRIPTION
**Pull request by vt-alt**
New tool to change files modification time to their last commit date.

### CODE REVIEW

File: Commands.md [hunk 1]
```diff
@@ -1495,3 +1496,11 @@ Opens the current git repository website in your default web browser.
 ```bash
 $ git browse
 ```
+
+## git utimes
+
+Change files modification time to their last commit date.
+
+```bash
+## git utimes
```
Comment (SuperSandro2000) [inline Commands.md:1505]:
```suggestion
$ git utimes
```

File: bin/git-utimes [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+    if [ -n "$t" ]; then
+      echo "+ touch -d $t $f" >&2
+      touch -d "$t" "$f"
+    fi
+  done
+  exit
+else
+  # `-n` should be limited or parallelization will not give effect,
+  # because all args will go into single worker.
+  NPROC=$(nproc)
```
Comment (SuperSandro2000) [inline bin/git-utimes:22]: That does not work on FreeBSD. See https://unix.stackexchange.com/questions/108821/number-of-cores-on-freebsd

**Review by SuperSandro2000 — CHANGES_REQUESTED** [review]

**Comment by vt-alt** [comment]
- Fixed `Commands.md`.
- Portable nproc oneliner is from https://stackoverflow.com/a/53427092/1478629
- Added additional safeguard in case `NPROC` is still null.

**Comment by vt-alt** [comment]
Ping! I fixed issues @SuperSandro2000 mentioned. Whats next?

File: Commands.md [hunk 3]
```diff
@@ -1495,3 +1496,11 @@ Opens the current git repository website in your default web browser.
 ```bash
 $ git browse
 ```
+
+## git utimes
+
+Change files modification time to their last commit date.
+
+```bash
+$ git utimes
```
Comment (spacewander) [inline Commands.md:1505]: Need to show the difference before / after running this command.

File: bin/git-utimes [hunk 4]
```diff
@@ -0,0 +1,25 @@
...
+#
+# Change files modification time to their last commit date
+#
+# Copyright (c) 2020 Vitaly Chikunov <vt@altlinux.org>.
+# SPDX-License-Identifier: MIT
+
+if [ "$1" = --touch ]; then
+  # Internal use option only just to parallelize things.
+  shift
+  for f; do
```
Comment (spacewander) [inline bin/git-utimes:11]: Is the for loop required?

File: bin/git-utimes [hunk 5]
```diff
@@ -0,0 +1,25 @@
...
+    t=$(git --no-pager log --no-renames --pretty=format:%cI -1 @ -- "$f" 2>/dev/null)
+    if [ -n "$t" ]; then
+      echo "+ touch -d $t $f" >&2
+      touch -d "$t" "$f"
+    fi
+  done
+  exit
```
Comment (spacewander) [inline bin/git-utimes:18]: The `exit` is redundant.

File: bin/git-utimes [hunk 6]
```diff
@@ -0,0 +1,25 @@
+#!/usr/bin/env bash
+#
+# Change files modification time to their last commit date
+#
+# Copyright (c) 2020 Vitaly Chikunov <vt@altlinux.org>.
+# SPDX-License-Identifier: MIT
```
Comment (spacewander) [inline bin/git-utimes:6]: Please don't add the Copyright and License. We already have and only have a LICENSE file.

**Review by spacewander — CHANGES_REQUESTED** [review]

File: bin/git-utimes (hunk 4 above)
Comment (vt-alt) [reply bin/git-utimes:11]: `for` is to support multiple arguments. Why not? This should reduce number of `exec`s. Are you against of it?

Comment (vt-alt) [reply bin/git-utimes:11]: Correction: s/execs/forks/

Comment (spacewander) [reply bin/git-utimes:11]: I got it.

Comment (spacewander) [reply bin/git-utimes:11]: @vt-alt
I got it, it is OK.

**Comment by vt-alt** [comment]
Force-pushed changes you requested - extended example in `Commands.md`, removed guard `exit`, removed license plate.

**Comment by vt-alt** [comment]
Almost three weeks passed since all requested changes are applied. Is there anything to add?

**Comment by spacewander** [comment]
@vt-alt 
Sorry for my delay!

**Review by spacewander — APPROVED** [review]

**Comment by vt-alt** [comment]
@spacewander Thanks!

**Pull request merged**